services/mailer/incoming/incoming_handler.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package incoming
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	issues_model "code.gitea.io/gitea/models/issues"
+	access_model "code.gitea.io/gitea/models/perm/access"
+	repo_model "code.gitea.io/gitea/models/repo"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/upload"
+	attachment_service "code.gitea.io/gitea/services/attachment"
+	issue_service "code.gitea.io/gitea/services/issue"
+	incoming_payload "code.gitea.io/gitea/services/mailer/incoming/payload"
+	"code.gitea.io/gitea/services/mailer/token"
+	pull_service "code.gitea.io/gitea/services/pull"
+)
+
+type MailHandler interface {
+	Handle(ctx context.Context, content *MailContent, user *user_model.User, payload []byte) error
```

Comment:
Please specify:
Is this `user` the `doer`, the `sender`, or the `recipient`?
Based on the code below, I guess it's the `sender`.
```suggestion
	Handle(ctx context.Context, content *MailContent, sender *user_model.User, payload []byte) error
```

Reply:
The `user` is the `doer` performing the reply action because he send the mail as `sender` in reply to the message he received as `recipient`. So `doer` should be acceptable.